exporter/awsemfexporter/datapoint_test.go

```diff
@@ -15,575 +15,402 @@
 package awsemfexporter
 
 import (
+	"fmt"
 	"reflect"
 	"testing"
 	"time"
 
-	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
-	"github.com/golang/protobuf/ptypes/wrappers"
+	aws "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/metrics"
 	"github.com/stretchr/testify/assert"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 	"go.uber.org/zap/zaptest/observer"
-
-	aws "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/metrics"
-	internaldata "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/opencensus"
 )
 
-func generateTestIntGauge(name string) *metricspb.Metric {
-	return &metricspb.Metric{
-		MetricDescriptor: &metricspb.MetricDescriptor{
-			Name: name,
-			Type: metricspb.MetricDescriptor_GAUGE_INT64,
-			Unit: "Count",
-			LabelKeys: []*metricspb.LabelKey{
-				{Key: "label1"},
-			},
-		},
-		Timeseries: []*metricspb.TimeSeries{
-			{
-				LabelValues: []*metricspb.LabelValue{
-					{Value: "value1", HasValue: true},
-				},
-				Points: []*metricspb.Point{
-					{
-						Value: &metricspb.Point_Int64Value{
-							Int64Value: 1,
-						},
-					},
-				},
-			},
-		},
-	}
+const instrLibName = "cloudwatch-otel"
+
+func generateTestGaugeMetric(name string, valueType string) pmetric.Metrics {
+	otelMetrics := pmetric.NewMetrics()
+	rs := otelMetrics.ResourceMetrics().AppendEmpty()
+	metrics := rs.ScopeMetrics().AppendEmpty().Metrics()
+	metric := metrics.AppendEmpty()
+	metric.SetName(name)
+	metric.SetUnit("Count")
+	gaugeMetric := metric.SetEmptyGauge()
+	gaugeDatapoint := gaugeMetric.DataPoints().AppendEmpty()
+	gaugeDatapoint.Attributes().PutStr("label1", "value1")
+
+	switch valueType {
+	case "double":
```

Comment:
Create a distinct type and constants for these.

```go
type gaugeValueType string

const doubleGauge gaugeValueType = "double"
```

and then only use the constants.  This should help avoid typos that accidentally change test behavior.

Reply:
Done !!! Thanks for your insight Anthony.